src/components/Swap/SwapButton/index.tsx

```diff
@@ -49,6 +50,7 @@ export default memo(function SwapButton({ disabled }: SwapButtonProps) {
   const deadline = useTransactionDeadline()
 
   const { type: wrapType, callback: wrapCallback } = useWrapCallback()
+  const switchNetworkAction = useSwitchNetwork(inputCurrency)
```

Comment:
technically chain !== network (is my understanding). since in web3-react, network refers to infura

Reply:
o did not know this